roles/pulp_common/tasks/install.yml

```diff
@@ -43,6 +43,7 @@
         shell: '{{ result.stdout.strip() }}'
         home: '{{ pulp_user_home }}'
         system: true
+        group: '{{ pulp_group }}'
```

Comment:
So this fixes the issue with new installers. Which is easy.

But what about existing installers? We have to design a solution, even if it means telling users to run `chown -R pulp:pulp /var/lib/pulp` .

We could recursively chown if we detect that the old version of Pulp was in a specific range. We currently recursively chown for the pulp 2 upgrade scenario.

Reply:
I thought about this too, and maybe we "just" pull that chmod out of the pulp2 block. But it might hit performance quite badly on big installations, so we may not want to run it more than once... Can we do something clever with a handler here?

Reply:
Seems like the change group recursively is not idempotent. I really do not understand why.
What happens if we do not change the group of existing files? Will pulp stop working properly, or will it "just" be a security concern (in this case a note to the user might be enough)?

Reply:
Update: Solved the idempotence issue by specifying `follow: false`.

Reply:
@mdellweg That's a good point. We could do a chmod on the dir itself (this task with recurse=false), and if it is changed, then repeat the task with "recurse=true"). We could do it with "when changed", or with a handler (since a handler is normally meant not to define config, but to make an applied config take effect.)

Reply:
@mikedep333 as i said, not following symlinks solves the idempotence.
There are static assets linked from the python packages that should not have their group changed anyway.

Reply:
So what we have right here works completely, I am just worried about performance. Hence my suggestion to repeat the task recursively if the dir needs to be changed initially.

We could always improve it later if users complain.